iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift

```diff
@@ -0,0 +1,123 @@
+                 "forgerock_oauth_scope": currentConfiguration.scopes.joined(separator: " "),
+                 "forgerock_cookie_name": currentConfiguration.cookieName ?? ""]
+            }
+            
+            
+            let discoveryURL = currentConfiguration.discoveryEndpoint
+            
+            let options = try await FROptions(config: config).discover(discoveryURL: discoveryURL)
+            return try await withCheckedThrowingContinuation({
+                (continuation: CheckedContinuation<Bool, Error>) in
+                do {
+                    try FRAuth.start(options: options)
+                    print("FRAuth SDK started using \(discoveryURL) discovery URL")
+                    FRLog.setLogLevel([.all])
+                    continuation.resume(returning: true)
+                } catch {
+                    print(String(describing: error))
+                    continuation.resume(throwing: error)
+                }
+            })
+        } else {
+            return false
+        }
+    }
+    
+    public func saveConfiguration() {
+        if let currentConfiguration = self.currentConfigurationViewModel {
+            let encoder = JSONEncoder()
+            let configuration = Configuration(clientId: currentConfiguration.clientId, scopes: currentConfiguration.scopes, redirectUri: currentConfiguration.redirectUri, signOutUri: currentConfiguration.signOutUri, discoveryEndpoint: currentConfiguration.discoveryEndpoint, environment: currentConfiguration.environment, cookieName: currentConfiguration.cookieName, browserType: currentConfiguration.getBrowserType())
+            if let encoded = try? encoder.encode(configuration) {
+                let defaults = UserDefaults.standard
+                defaults.set(encoded, forKey: "CurrentConfiguration")
+            }
+        }
+    }
+    
+    public func deleteSavedConfiguration() {
+        let defaults = UserDefaults.standard
+        defaults.removeObject(forKey: "CurrentConfiguration")
+    }
+    
+    public func defaultConfigurationViewModel() -> ConfigurationViewModel {
+        let defaults = UserDefaults.standard
+        if let savedConfiguration = defaults.object(forKey: "CurrentConfiguration") as? Data {
+            let decoder = JSONDecoder()
+            if let loadedConfiguration = try? decoder.decode(Configuration.self, from: savedConfiguration) {
+                return ConfigurationViewModel(clientId: loadedConfiguration.clientId, scopes: loadedConfiguration.scopes, redirectUri: loadedConfiguration.redirectUri, signOutUri: loadedConfiguration.signOutUri, discoveryEndpoint: loadedConfiguration.discoveryEndpoint, environment: loadedConfiguration.environment, cookieName: loadedConfiguration.cookieName, browserSeletorType: BrowserSeletorTypes(rawValue: loadedConfiguration.browserType.description) ?? BrowserSeletorTypes.authSession)
+            }
+        }
+        
+        return ConfigurationViewModel(
```

Comment:
I would suggest to add a TODO here. Something like:

`//TODO: Provide here the client configuration. Replace the placeholder with the OAuth2.0 client details`